common/countdown/exp_duration.go

```diff
@@ -0,0 +1,78 @@
+// A countdown timer that will mostly be used by XDPoS v2 consensus engine
+package countdown
+
+import (
+	"math"
+	"time"
+
+	"github.com/XinFinOrg/XDPoSChain/core/types"
+	"github.com/XinFinOrg/XDPoSChain/log"
+)
+
+const max_exponent_upperbound uint8 = 32
+
+type ExpTimeoutDuration struct {
+	duration     time.Duration
+	base         float64
+	max_exponent uint8
+}
+
+func NewExpTimeoutDuration(duration time.Duration, base float64, max_exponent uint8) *ExpTimeoutDuration {
+	d := &ExpTimeoutDuration{
+		duration:     duration,
+		base:         base,
+		max_exponent: max_exponent,
+	}
+	d.sanityCheck()
+	return d
+}
+
+func (d *ExpTimeoutDuration) sanityCheck() {
+	if d.max_exponent >= max_exponent_upperbound {
+		log.Error("max_exponent (e)= >= max_exponent_upperbound (e_upper)", "e", d.max_exponent, "e_upper", max_exponent_upperbound)
+		panic("max_exponent (e)= >= max_exponent_upperbound (e_upper)")
```

Comment:
`log.Crit` can be used here instead of `log.Error` and `panic`. But I think it's not a good idea to call log function in the common package.